src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1839,6 +1839,31 @@
     <Field Name="WasTargetTyped" Type="bool" PropertyOverrides="true"/>
   </Node>
   
+  <!--
+  A target-typed collection creation expression.
+  It does not survive past initial binding.
+  -->
+  <Node Name="BoundUnconvertedCollectionLiteralExpression" Base="BoundExpression">
+    <!-- Type is not significant for this node type; always null -->
+    <Field Name="Type" Type="TypeSymbol?" Override="true" Null="always"/>
+    <Field Name="Binder" Type="Binder" Null="disallow"/>
+  </Node>
+
+  <Node Name="BoundCollectionLiteralExpression" Base="BoundExpression">
+    <!-- Non-null type is required for this node kind -->
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="Constructor" Type="MethodSymbol" Null="disallow"/>
```

Comment:
interesting.  feels like this could be null in some cases.  For example, if we use `default(SomeStruct)` to initialize things.